### DISCUSSION TITLE
Enable per segment color for LINE_STRIP points/image annotation

### DISCUSSION THREAD

**Discussion by vivek-rane** [foxglove/community#715] [Q&A] [answered]
Right now only POINTS and LINE_LIST is supported for per segment color. It would be nice to extend the same to LINE_STRIP and LINE_LOOP.

e.g. The following 6 points should produce a line with 5 segments with the first 5 colors in the list
```
{
    timestamp: event.receiveTime,
    type: PointAnnotationTypes.LINE_STRIP,
    thickness: 2,
    points: [
      { x: 500.1, y: 100.5 },
      { x: 200.2, y: 200.632322 },
      { x: 300.2, y: 200.632322 },
      { x: 200.2, y: 100.632322 },
      { x: 100.2, y: 200.632322 },
      { x: 50.2, y: 200.632322 },
    ],
    outline_color: { r: 0, g: 0, b: 0, a: 1 },
    outline_colors: [
      { r: 0, g: 0, b: 1, a: 1 },
      { r: 0, g: 1, b: 1, a: 1 },
      { r: 1, g: 0, b: 1, a: 1 },
      { r: 0, g: 0, b: 1, a: 1 },
      { r: 1, g: 1, b: 1, a: 1 },
      { r: 1, g: 1, b: 0, a: 1 },
    ], 
    fill_color: { r: 0, g: 0, b: 0, a: 0 },
  }
```
Here, changing POINTS to LINE_STRIP should work with the color list (right now the line just shows up as black):
![image](https://github.com/foxglove/community/assets/25373098/090d437c-960c-45e9-b4d0-0957bcb473df)

Side note: The documentation is a bit inadequate on this front. It is super confusing about:
1. Whether `LINE_STRIP` and `LINE_LOOP` are supported with `outline_colors`
2. When `outline_color` is used and when `outline_colors` are used for the drawing.
3. What the required points to colors ratio is (1:1 or 1:2) for `LINE_STRIP` and what happens if that ratio isn't correct.

**Comment by jtbandes** [comment]
Internal tracking ticket: FG-3505

**Comment by jtbandes** [comment, marked as answer]
This is done in https://github.com/foxglove/studio/pull/6973 and will be available in the next release!
